docs/scenarios_authoring/Spel.md

```diff
@@ -276,27 +276,28 @@ More usage examples can be found in [this section](#conversions-between-datetime
 
 #### Explicit conversions
 
-Explicit conversions are available in utility classes and build-in java conversion mechanisms:
-
-| Expression                                                      | Result                     | Type            |
-|-----------------------------------------------------------------|----------------------------|-----------------|
-| `#NUMERIC.toNumber('42')`                                       | 42                         | Number          |
-| `#NUMERIC.toNumber('42').toString()`                            | '42'                       | String          |
-| `#DATE_FORMAT.parseOffsetDateTime('2018-10-23T12:12:13+00:00')` | 1540296720000              | OffsetDateTime  |
-| `#DATE_FORMAT.parseLocalDateTime('2018-10-23T12:12:13')`        | 2018-10-23T12:12:13+00:00  | LocalDateTime   |
-
-### Casting
-
-When a type cannot be determined by parser, the type is presented as `Unknown`. When we know what the type will be on
-runtime, we can cast a given type, and then we can operate on the cast type.
-
-E.g. having a variable `obj` of a type: `List[Unknown]` and we know the elements are strings then we can cast elements
-to String: `#obj.![#this.castToOrNull('String')]`.
-
-Available methods:
-- `canCastTo` - checks if a type can be cast to a given class.
-- `castTo` - casts a type to a given class or throws exception if type cannot be cast.
-- `castToOrNull` - casts a type to a given class or return null if type cannot be cast.
+Explicit conversions are available in utility classes, build-in java conversion mechanisms and via extension methods:
+
+| Expression                                                      | Result                     | Type          |
+|-----------------------------------------------------------------|----------------------------|---------------|
+| `#NUMERIC.toNumber('42')`                                       | 42                         | Number        |
+| `#NUMERIC.toNumber('42').toString()`                            | '42'                       | String        |
+| `#DATE_FORMAT.parseOffsetDateTime('2018-10-23T12:12:13+00:00')` | 1540296720000              | OffsetDateTime|
+| `#DATE_FORMAT.parseLocalDateTime('2018-10-23T12:12:13')`        | 2018-10-23T12:12:13+00:00  | LocalDateTime |
+| `'true'.toBoolean`                                              | true                       | Boolean       |
+| `'11'.toLong`                                                   | 11                         | Long          |
+| `'1.1'.toDouble`                                                | 1.1                        | Double        |
+| `2.1.toBigDecimal`                                              | 2.1                        | BigDecimal    |
+| `{a: 1, b: 2}.keySet.toList`                                    | {'a', 'b'}                 | List[String]  |
+
+Functions which start with `to` are extension methods which allows convert value to appropriate type. 
```

Comment:
We should describe what happen when this conversion fail. Also, I see some chaos in this description. We write about `to`, then we jump `toLong` which is shown as an example but without explanation of all similar examples, then we back to `to`-like methods (`is`, `toOrNull`).

Reply:
We agreed on that, that I will adjust docs in next PR